spring-cloud-gcp-data-spanner/src/test/java/com/google/cloud/spring/data/spanner/core/SpannerTemplateTests.java

```diff
@@ -342,65 +338,62 @@ public void readWriteTransactionPartitionedDmlTest() {
               TransactionCallable transactionCallable = invocation.getArgument(0);
               return transactionCallable.run(transactionContext);
             });
+    Function<SpannerTemplate, String> testSpannerOperations = spannerTemplate -> {
+      spannerTemplate.executePartitionedDmlStatement(Statement.of("DML statement here"));
+      return "all done";
+    };
 
-    this.spannerTemplate.performReadWriteTransaction(
-        spannerTemplate -> {
-          spannerTemplate.executePartitionedDmlStatement(Statement.of("DML statement here"));
-          return "all done";
-        });
+    assertThatThrownBy(() -> this.spannerTemplate.performReadWriteTransaction(testSpannerOperations))
+            .hasMessage("A read-write transaction template cannot execute partitioned" + " DML.");
   }
 
   @Test
-  public void nullDatabaseClientTest() {
-
-    this.expectedException.expect(IllegalArgumentException.class);
-    this.expectedException.expectMessage("A valid database client for Spanner is required.");
+  void nullDatabaseClientTest() {
 
-    new SpannerTemplate(
-        null, this.mappingContext, this.objectMapper, this.mutationFactory, this.schemaUtils);
+    assertThatThrownBy(() -> new SpannerTemplate(
+            null, this.mappingContext, this.objectMapper, this.mutationFactory, this.schemaUtils))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("A valid database client for Spanner is required.");
   }
 
   @Test
-  public void nullMappingContextTest() {
+  void nullMappingContextTest() {
 
-    this.expectedException.expect(IllegalArgumentException.class);
-    this.expectedException.expectMessage("A valid mapping context for Spanner is required.");
-
-    new SpannerTemplate(
-        () -> this.databaseClient, null, this.objectMapper, this.mutationFactory, this.schemaUtils);
+    assertThatThrownBy(() ->  new SpannerTemplate(
+            () -> this.databaseClient, null, this.objectMapper, this.mutationFactory, this.schemaUtils))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("A valid mapping context for Spanner is required.");
   }
 
   @Test
-  public void nullObjectMapperTest() {
-
-    this.expectedException.expect(IllegalArgumentException.class);
-    this.expectedException.expectMessage("A valid entity processor for Spanner is required.");
+  void nullObjectMapperTest() {
 
-    new SpannerTemplate(
-        () -> this.databaseClient,
-        this.mappingContext,
-        null,
-        this.mutationFactory,
-        this.schemaUtils);
+    assertThatThrownBy(() ->  new SpannerTemplate(
+            () -> this.databaseClient,
+            this.mappingContext,
+            null,
+            this.mutationFactory,
+            this.schemaUtils))
+            .isInstanceOf(IllegalArgumentException.class)
```

Comment:
nit: Could you reformat this paragraph, so that the `isInstanceOf()` indentation is smaller than "carryover line" indentation? The easiest way is probably to split `))` to two different lines.